lib/internal/test_runner/test.js

```diff
@@ -209,6 +210,19 @@ class TestContext {
     return subtest.start();
   }
 
+  retries(number) {
+    validateNumber(number, 'number', 0, NumberMAX_SAFE_INTEGER);
+    this.#test.retries = number;
+  }
+
+  get retryCount() {
```

Comment:
I'm not sure that we need `retryCount()`. We don't really have getters and setters for the other options and I'd like to avoid API bloat. I think we should try landing this initially without this API and see if people request it (I doubt it will be needed because people know what value they have set). I could also picture people being confused by `retryCount` and `currentAttempt`.